ftprime/argrecorder.py

```diff
@@ -196,6 +208,34 @@ def get_nodes(self, input_ids):
         """
         return [self.node_ids[j] for j in input_ids]
 
+    def add_individuals(self, input_ids, times, flagss=None, populations=None):
+        '''
+        Add new individuals, resulting in records as in `.add_individual`, but
+        more efficiently.
+
+        :param iterable of int input_ids: The input ID of the new individual.
+        :param iterable of float times: The time of birth of the individual.
+        :param iterable of int flagss: Any msprime flags to record.
+        :param iterable of int populations: The population ID of birth of the
+            individual.
+        '''
+        if flagss is None:
+            flagss = itertools.cycle((msprime.NODE_IS_SAMPLE, ))
+        if populations is None:
+            populations = itertools.cycle((msprime.NULL_POPULATION, ))
+
+        # inflate to a list so we have a length; then set up the map from
+        # node_ids to tree sequence samples
+        input_ids = list(self._valid_input_ids(input_ids))
+        for i, n in zip(input_ids, itertools.count(start=self.nodes.num_rows)):
+            self.node_ids[i] = n
+        nt = np.fromiter(zip(flagss, times, populations), dtype=node_dt,
```

Comment:
Is there an efficiency advantage to doing this rather than
```
np_flags = np.fromiter(flagss, ...)
np_times = np.fromiter(times, ...)
np_pops = np.fromiter(populations, ...)
self.__nodes.append_columns(flags=np_flags, time=np_times, population=np_pops)
```
... ?  This way (with `zip()`, etc) seems possibly less efficient, and harder to read anyhow.  Hm, I guess it only has to go and look and allocate a table once, instead of three vectors?